test/org/sagebionetworks/bridge/services/AuthenticationServiceImplTest.java

```diff
@@ -154,7 +154,18 @@ public void canResendEmailVerification() throws Exception {
         try {
             Email email = new Email(testUser.getStudyIdentifier(), user.getEmail());
             authService.resendEmailVerification(user.getStudyIdentifier(), email);
-        } catch (ConsentRequiredException e) {
+        } finally {
+            helper.deleteUser(user);
+        }
+    }
+    
+    @Test
+    public void resendEmailVerificationLooksSuccessfulWhenNoAccount() throws Exception {
```

Comment:
No, you can reset password even before you've consented. Someone like a developer who never consents to the study can reset their password.

Regarding the verifyEmail and resetPassword callbacks: both of these take an sptoken and if the token is invalid, Stormpath returns a 404 (could not find the sptoken). Since this is all you get back if you use these calls, they don't leak information about study participation. And since you need a valid token to get past that point, the next message you'll always see is a success since by definition, you've got a token that was triggered by a valid account.